apps/analog-app/src/app/pages/form-action.directive.ts

```diff
@@ -0,0 +1,40 @@
+import { Directive, inject, input, output } from '@angular/core';
+import { Router } from '@angular/router';
+
+export type ActionResult = {
+  type: string;
+};
+
+@Directive({
+  selector: 'form[action],form[method]',
+  host: {
+    '(submit)': `submitted($event)`,
+  },
+  standalone: true,
+})
+export class FormActionDirective {
+  action = input<string>('');
+  onSuccess = output<unknown>();
+  onError = output<unknown>();
+  state = output<'submitting' | 'error'>();
+  router = inject(Router);
+
+  submitted($event: { target: HTMLFormElement } & Event) {
+    $event.preventDefault();
+    const body = new FormData($event.target);
```

Comment:
I didn't know we could do that 🤦 I was using viewChild for this as I couldn't find this info within `$event` 🤷